electrum/constants.py

```diff
@@ -39,136 +41,241 @@ def read_json(filename, default):
         r = default
     return r
 
-
-GIT_REPO_URL = "https://github.com/spesmilo/electrum"
-GIT_REPO_ISSUES_URL = "https://github.com/spesmilo/electrum/issues"
+GIT_REPO_URL = "https://github.com/BTCGPU/electrum"
+GIT_REPO_ISSUES_URL = GIT_REPO_URL + "/issues"
 BIP39_WALLET_FORMATS = read_json('bip39_wallet_formats.json', [])
 
 
 class AbstractNet:
 
     BLOCK_HEIGHT_FIRST_LIGHTNING_CHANNELS = 0
 
+    POW_TARGET_SPACING = 10 * 60
+    POW_TARGET_TIMESPAN_LEGACY = 14 * 24 * 60 * 60
+
+    DIGI_AVERAGING_WINDOW = 30
+    DIGI_MAX_ADJUST_DOWN = 32
+    DIGI_MAX_ADJUST_UP = 16
+
+    LWMA_AVERAGING_WINDOW = 45
+    LWMA_ADJUST_WEIGHT_LEGACY = 13772
+    LWMA_ADJUST_WEIGHT = 13772
+    LWMA_MIN_DENOMINATOR_LEGACY = 10
+    LWMA_MIN_DENOMINATOR = 10
+    LWMA_SOLVETIME_LIMITATION = True
+
+    EQUIHASH_PARAMS = EquihashParams(n=200, k=9, personalization=b'ZcashPoW')
+    EQUIHASH_PARAMS_FORK = EquihashParams(n=144, k=5, personalization=b'BgoldPoW')
+
+    HEADER_SIZE_ORIGIN = 80
+    HEADER_SIZE_LEGACY = 141
+    
+    CHUNK_SIZE = 252
+
+    SIGHASH_FORK_BTG = 0x4f40
+
     @classmethod
     def max_checkpoint(cls) -> int:
         return max(0, len(cls.CHECKPOINTS) * 2016 - 1)
 
     @classmethod
-    def rev_genesis_bytes(cls) -> bytes:
-        return bytes.fromhex(bitcoin.rev_hex(cls.GENESIS))
+    def get_checkpoint_hash(cls, height) -> str:
+        def is_height_checkpoint():
+            within_cp_range = height <= cls.max_checkpoint()
+            at_chunk_boundary = (height+1) % 2016 == 0
+            return within_cp_range and at_chunk_boundary
 
+        if is_height_checkpoint():
+            index = height // 2016
+            h, t = cls.CHECKPOINTS[index]
+            return h
 
-class BitcoinMainnet(AbstractNet):
+    @classmethod
+    def rev_genesis_bytes(cls) -> bytes:
+        return bytes.fromhex(bitcoin.rev_hex(cls.GENESIS))
 
+class BitcoinGoldMainnet(AbstractNet):
     TESTNET = False
     WIF_PREFIX = 0x80
-    ADDRTYPE_P2PKH = 0
-    ADDRTYPE_P2SH = 5
-    SEGWIT_HRP = "bc"
+    ADDRTYPE_P2PKH = 38
+    ADDRTYPE_P2SH = 23
+    SEGWIT_HRP = "btg"
+    HEADERS_URL = "https://headers.bitcoingold.org/blockchain_headers.gz"
```

Comment:
Btw, do you know how to generate this file?

Reply:
I think that is the answer.. https://github.com/spesmilo/electrum/issues/5469. We just have to install ElectrumG with an empty checkpoints.json file.

Reply:
![2](https://user-images.githubusercontent.com/23510057/166437165-8edb3a44-9cfd-4e66-ad6a-9410a3d34c22.PNG)
![4](https://user-images.githubusercontent.com/23510057/166437178-7026f79e-030a-42d6-b423-1690011a5c13.PNG)
![5](https://user-images.githubusercontent.com/23510057/166437198-e0e1c0ef-c213-4564-beaf-e782a17e989e.PNG)
![6](https://user-images.githubusercontent.com/23510057/166437210-62664019-6303-4ece-a13c-fd3659956cf3.PNG)
@h4x3rotab @R-Santev @tolak @martin-key